src/Services/WordWrapper.vala

```diff
@@ -0,0 +1,94 @@
+
+/*
+* Copyright (c) 2017 Felipe Escoto (https://github.com/Philip-Scott/Notes-up)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 2 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+* Boston, MA 02111-1307, USA.
+*
+* Authored by: Felipe Escoto <felescoto95@hotmail.com>
+*/
+
+/**
+ * This class handles the wrapping logic for a given text within the bounds of 
+ * two informed strings, which act as "string wrappers" for the text 
+ */
+public class WordWrapper : Object {
+    private string text;
+
+    private int n_leading_spaces = 0;
+    private int n_trailing_spaces = 0;
+
+    private static Regex regex_start_whitespaces;
+    private static Regex regex_end_whitespaces;
+
+    class construct {
+        //  TODO: create 1 single regex to capture both starting and ending whitespaces only
+        const string START_WHITESPACES_PATTERN = "^(\\s)+"; 
+        const string END_WHITESPACES_PATTERN = "(\\s)+$"; 
```

Comment:
constants go on the top of the class, just below `public class WordWrapper : Object {`